specification/hardwaresecuritymodules/HardwareSecurityModules.Management/DedicatedHsm.tsp

```diff
@@ -0,0 +1,143 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.HardwareSecurityModules;
+/**
+ * Resource information with extended details.
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance"
+#suppress "@azure-tools/typespec-azure-core/no-private-usage"
+@Azure.ResourceManager.Private.armResourceInternal(DedicatedHsmProperties)
+@TypeSpec.Http.Private.includeInapplicableMetadataInPayload(false)
+model DedicatedHsm extends Foundations.TrackedResource {
+  ...ResourceNameParameter<
+    Resource = DedicatedHsm,
+    KeyName = "name",
+    SegmentName = "dedicatedHSMs"
+  >;
+
+  /**
+   * SKU details
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "For backward compatibility"
+  sku: Sku;
+
+  ...Azure.ResourceManager.AvailabilityZonesProperty;
+
+  /**
+   * Properties of the dedicated HSM
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property"
+  @Azure.ResourceManager.Private.conditionalClientFlatten
+  @Azure.ResourceManager.Private.armResourcePropertiesOptionality(false)
+  properties: DedicatedHsmProperties;
+}
+
+@armResourceOperations
+interface DedicatedHsms {
```

Comment:
Nit: Should this interface name match the prefix used in the operationIds of the routes within it?
```suggestion
interface DedicatedHsm {
```

Reply:
Maybe, @melina5656 check if all the operations in this interface have an operationid starts with `DedicatedHsm_` , if so, try to rename to check if there is name conflict.

Reply:
This is handled by `@clientLocation`